assets/js/amp-customize-controls.js

```diff
@@ -0,0 +1,194 @@
+			if ( ampToggle && ! ampEnabled ) {
+				toggleControl( false, true, true );
+			} else if ( ! ampToggle && ! ampEnabled ) {
+
+				// Enable toggle but don't trigger anything else.
+				$( '.amp-toggle input' ).prop( 'disabled', false );
+			}
+		}
+
+		/**
+		 * Central function to control UI toggle elements.
+		 *
+		 * @param {bool} checked If toggle input should be checked.
+		 * @param {bool} disabled If toggle input should be disabled.
+		 * @param {bool} notification If tooltip notification should appear.
+		 * @return {void}
+		 */
+		function toggleControl( checked, disabled, notification ) {
+			var $input = $( '.amp-toggle input' ),
+				$tooltip = $( '.amp-toggle .tooltip' ),
+				tooltipTimer = 5000;
+
+			$input.prop( 'checked', checked );
+			$input.prop( 'disabled', disabled );
+
+			if ( notification ) {
+				$tooltip.fadeIn();
+				setTimeout( function() {
+					$tooltip.fadeOut();
+				}, tooltipTimer );
+			} else {
+				$tooltip.hide();
+			}
+
+			$input.trigger( 'change' );
+		}
+
+		// AMP panel triggers the input toggle for AMP preview.
+		panel.expanded.bind( function() {
+			var panelOpen = panel.expanded.get();
+			toggleControl( panelOpen, panelOpen, false );
+		} );
+
+		// If AMP panel is open, let it handle enable/disable of the toggle,
+		// otherwise auto enable toggle when mobile device is selected.
+		api.previewedDevice.bind( function( device ) {
+			if ( ! panel.expanded.get() && ! ampToggle ) {
+				toggleControl( 'mobile' === device, false, false );
+			}
+		} );
+
+		// Preview is ready, check if it's AMP'd in case we need to show tooltip.
+		api.previewer.bind( 'ready', isPageAmpified );
+
+		// Persist the presence or lack of the amp=1 param when navigating in the preview,
+		// even if current page is not yet supported.
+		api.previewer.previewUrl.validate = ( function( prevValidate ) {
+			return function( value ) {
+				var val = prevValidate.call( this, value );
+				if ( val ) {
+					val = setCurrentAmpUrl( val );
+				}
+				return val;
+			};
+		} )( api.previewer.previewUrl.validate );
+
+		// Adding checkbox toggle before device selection.
+		$( '.devices-wrapper' ).before( '<label class="amp-toggle">' +
+			'<input type="checkbox">' +
+			'<span class="slider"></span>' +
+		'</label>' );
+
+		// Notification tooltip for AMP toggle.
+		$( '.amp-toggle input' ).before( '<span class="tooltip">' +
+			'This page is not AMP compatible.<br>' +
+			'<a data-post="' + window.ampPost + '">Navigate to an AMP compatible page</a>' +
+		'</span>' );
+
+		$( '.amp-toggle .tooltip a' ).on( 'click', function() {
+			var url = $( this ).data( 'post' );
+			if ( url.length ) {
+				api.previewer.previewUrl.set( setCurrentAmpUrl( url ) );
+				toggleControl( false, false, false );
+			}
+		} );
+
+		// Main control for toggling AMP preview.
+		$( '#customize-footer-actions' ).on( 'change', '.amp-toggle input', function( event ) {
+			ampToggle = $( this ).is( ':checked' );
```

Comment:
I think the state here should be moved to `wp.customize.state`. For example, at the top of this file do:

```js
wp.customize.state.add( 'ampEnabled', new wp.customize.Value() );
wp.customize.state.add( 'ampAvailable', new wp.customize.Value() );
```

The `ampEnabled` state can manage whether the toggle is on or off, whereas the `ampAvailable` state can manage whether or not AMP is available in the preview. Then it can be populated via messages from the preview via:

```js
wp.customize.bind( 'ready', function( available ) {
    wp.customize.previewer.bind( 'ampAvailable', function( available ) {
        wp.customize.state( 'ampAvailable' ).set( available );
    } );
} );
```

Then the `disabled` state of the toggle can be managed by the state like so:

```js
wp.customize.state( 'ampAvailable' ).bind( function( available ) {
     $( '.amp-toggle input' ).prop( 'disabled', available );
} );
```

As noted above about sending messages from the preview about whether AMP is supported, there can then be a message receiver that then updates this state. In the preview it could do:

```js
var ampAvailable = <?php echo wp_json_encode( is_singular() && post_supports_amp( get_queried_object() ) ); ?>;
//...
wp.customize.bind( 'preview-ready', function() {
    wp.customize.preview.send( 'ampAvailable', ampAvailable );
} );
```

Clicking on the checkbox can just change the `ampEnabled`state, something like this:

```js
$( '.amp-toggle input' ).on( 'click', function( event ) {
    event.preventDefault();
    wp.customize.state( 'ampEnabled' ).set( ! wp.customize.state( 'ampEnabled' ).get() );
} );
```

And likewise the state of the button then can then in turn be managed by this state as well, as well as then setting the `previewUrl`:

```js
wp.customize.state( 'ampEnabled' ).bind( function( enabled ) {
    $( '.amp-toggle input' ).prop( 'checked', enabled );
    toggleAmpView();
} );
```